### ISSUE TITLE
Problem with content reload when placing a new dynamic function

### ISSUE THREAD

**Issue by kartobi** [alkacon/opencms-core#484]
After updating from OpenCms 9.0.1 to 10.5.1 the automatic reload of an content element isn't working any more.

Drag&Drop of a new dynamic function logs the following message:
```
03 Mär 2017 15:18:29,205 [http-nio-8443-exec-2] INFO  org.opencms.workplace.explorer.CmsResourceUtil  - org.opencms.file.CmsVfsResourceNotFoundException: Error reading resource from path "/~".
03 Mär 2017 15:18:29,229 [http-nio-8443-exec-2] WARN  org.opencms.util.CmsNullIgnoringConcurrentMap  - Invalid null value in map
java.lang.Exception
	at org.opencms.util.CmsNullIgnoringConcurrentMap.put(CmsNullIgnoringConcurrentMap.java:172)
	at org.opencms.util.CmsNullIgnoringConcurrentMap.putAll(CmsNullIgnoringConcurrentMap.java:196)
	at org.opencms.xml.containerpage.CmsContainerElementBean.initSettings(CmsContainerElementBean.java:575)
	at org.opencms.ade.containerpage.CmsElementUtil.getContentByContainer(CmsElementUtil.java:888)
	at org.opencms.ade.containerpage.CmsElementUtil.getContentsByContainerName(CmsElementUtil.java:918)
	at org.opencms.ade.containerpage.CmsElementUtil.getElementData(CmsElementUtil.java:527)
	at org.opencms.ade.containerpage.CmsContainerpageService.getNewElement(CmsContainerpageService.java:2450)
	at org.opencms.ade.containerpage.CmsContainerpageService.getNewElementData(CmsContainerpageService.java:981)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:587)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:333)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:303)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:373)
	at com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.opencms.gwt.CmsGwtService.service(CmsGwtService.java:194)
	at org.opencms.main.OpenCmsCore.invokeGwtService(OpenCmsCore.java:1722)
	at org.opencms.main.OpenCmsServlet.doGet(OpenCmsServlet.java:158)
	at org.opencms.main.OpenCmsServlet.doPost(OpenCmsServlet.java:177)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.opencms.jsp.jsonpart.CmsJsonPartFilter.doFilter(CmsJsonPartFilter.java:281)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:616)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:509)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1104)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:684)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1520)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1476)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
03 Mär 2017 15:18:29,232 [http-nio-8443-exec-2] WARN  org.opencms.util.CmsNullIgnoringConcurrentMap  - Invalid null value in map
java.lang.Exception
	at org.opencms.util.CmsNullIgnoringConcurrentMap.put(CmsNullIgnoringConcurrentMap.java:172)
	at org.opencms.util.CmsNullIgnoringConcurrentMap.putAll(CmsNullIgnoringConcurrentMap.java:196)
	at org.opencms.xml.containerpage.CmsContainerElementBean.initSettings(CmsContainerElementBean.java:575)
	at org.opencms.ade.containerpage.CmsElementUtil.getContentByContainer(CmsElementUtil.java:888)
	at org.opencms.ade.containerpage.CmsElementUtil.getContentsByContainerName(CmsElementUtil.java:918)
	at org.opencms.ade.containerpage.CmsElementUtil.getElementData(CmsElementUtil.java:527)
	at org.opencms.ade.containerpage.CmsContainerpageService.getNewElement(CmsContainerpageService.java:2450)
	at org.opencms.ade.containerpage.CmsContainerpageService.getNewElementData(CmsContainerpageService.java:981)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:587)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:333)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:303)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:373)
	at com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.opencms.gwt.CmsGwtService.service(CmsGwtService.java:194)
	at org.opencms.main.OpenCmsCore.invokeGwtService(OpenCmsCore.java:1722)
	at org.opencms.main.OpenCmsServlet.doGet(OpenCmsServlet.java:158)
	at org.opencms.main.OpenCmsServlet.doPost(OpenCmsServlet.java:177)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.opencms.jsp.jsonpart.CmsJsonPartFilter.doFilter(CmsJsonPartFilter.java:281)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:616)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:509)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1104)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:684)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1520)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1476)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
```

Everything works fine when choosing a function provider and Save&Exit - the content of referenced jsp is displayed on page.

When editing the dynamic function directly again now (without page reload) e.g. choose a different function provider, the changes are not shown. A page reload is necessary.

**Comment by patric-dosch-vi** [comment]
any idea here?

I have posted a question on the list http://opencms.996256.n3.nabble.com/Error-when-place-a-new-module-org-opencms-util-CmsNullIgnoringConcurrentMap-Invalid-null-value-in-map-td26727.html and we found out that the same happend with the opencms internal content type "dynamic function" - so it seems not to be a configuration problem of our content types.

Regards, Patric

**Comment by tHerrmann** [comment]
I think the commit above should fix the issue. 
The exception logged by CmsNullIgnoringConcurrentMap is not related to the problem and should not be written into the WARN channel.

**Comment by tHerrmann** [comment]
The fix to this issue has been released with version [10.5.2](https://github.com/alkacon/opencms-core/releases/tag/build_10_5_2).
